### ISSUE TITLE
vmctl too many tcp connections

### ISSUE THREAD

**Issue by renchili** [VictoriaMetrics/VictoriaMetrics#4954]
### Describe the bug

When i running vmctl, vminsert, vmselect to migrate data

all 3process are running in 1 pod with 3 containers

![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/33677006/f05cc6f7-cebc-4e74-b638-9eece834e5ab)

There are too many random ports opened but not closed ,then it caused random ports exausted.

### To Reproduce

filter: match[]={job="SoTest-Conductor", instance=~"10.123.2.52:8080"}
start: 2023-07-01T00:00:00Z
2023/09/05 09:27:23 Exploring metrics...
2023/09/05 09:27:56 Selected time range will be split into 1594 ranges according to "hour" step. Requests to make: 103610

 total bytes: 4.3 GB;
  bytes/s: 7.4 MB;
  requests: 103610;
  requests retries: 10;
  
  

### Version

vmselect-20230630-162911-tags-v1.91.3-cluster-0-g12f262c331
vmctl version vmctl-20230630-132322-tags-v1.91.3-0-g7226242070
vminsert-20230630-162908-tags-v1.91.3-cluster-0-g12f262c331

### Used command-line flags

vmctl
```
./vmctl-prod vm-native  {{- include "victoria-metric.vmselect-pod-fqdn" . }}  {{- include "victoria-metric.vminsert-pod-fqdn" . }} \
              --vm-native-filter-time-start=2023-07-01T00:00:00Z  \
              --vm-native-step-interval=hour  \
              --vm-concurrency=2000  \
              --verbose=true  \
              --vm-native-filter-match='{job="SoTest-Conductor", instance=~"xxxxxxx:8080"}' \
              --vm-extra-label source_vm=vmctl-ssk
```

vmselect
```
./vmselect-prod {{- include "victoria-metric.vmselect-source-vmstorage" . }} \
              -dedup.minScrapeInterval=13s  \
              -search.maxQueueDuration=500s  \
              -search.logSlowQueryDuration=15s \
              -search.maxQueryDuration=300s \
              -search.maxQueryLen=10MB \
              -search.maxStalenessInterval=120s \
              -search.minStalenessInterval=120s \
              -search.maxSamplesPerQuery=5000000000 \
              -search.maxSeries=3000000 \
              -search.maxConcurrentRequests=24 \
              -vmstorageDialTimeout=30s \
              -search.setLookbackToStep=true
```
vminsert
```
./vminsert-prod {{- include "victoria-metric.vminsert-target-vmstorage" . }} \
              -maxConcurrentInserts=32 \
              -maxLabelsPerTimeseries=50
```

**Comment by hagen1778** [comment]
Hello @renchili!

Can you try passing `-vm-native-disable-http-keep-alive` cmd-line flag to vmctl? I'd also recommend changing `--vm-native-step-interval` to `day` or `week`, as it will reduce amount of requests to make by 24 and 168 respectively. Please note, default `--vm-native-step-interval` is set to 1 month.

**Comment by renchili** [comment]
> Hello @renchili!
> 
> Can you try passing `-vm-native-disable-http-keep-alive` cmd-line flag to vmctl? I'd also recommend changing `--vm-native-step-interval` to `day` or `week`, as it will reduce amount of requests to make by 24 and 168 respectively. Please note, default `--vm-native-step-interval` is set to 1 month.

cannot by day the data by day exceed the max time series limitation set on vmselectand vmstorage

**Comment by hagen1778** [comment]
> cannot by day the data by day exceed the max time series limitation set on vmselect and vmstorage

You can increase the limitation on vmselect if you have enough resources for processing. Please note, the default complexity limits are usually static and might need to be adjusted based on workload and amount of available resources.

Have you tried other recommendations?

**Comment by renchili** [comment]
> > cannot by day the data by day exceed the max time series limitation set on vmselect and vmstorage
> 
> You can increase the limitation on vmselect if you have enough resources for processing. Please note, the default complexity limits are usually static and might need to be adjusted based on workload and amount of available resources.
> 
> Have you tried other recommendations?

Thank you for more information. Well, I do not understand what the word `enough` resources stands for.
Could you enlarge this point? how much should it be? 
For your information this is our cluster size:
![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/33677006/1801fed7-08fb-4ca3-a3c3-dba5b723581c)

**Comment by makasim** [comment]
@zekker6 could metrics introduced in https://github.com/VictoriaMetrics/VictoriaMetrics/pull/10276 help investigate cases like this ?

**Comment by zekker6** [comment]
@makasim Yes, for sure. Metrics in vmctl will cover amount of HTTP requests as well as amount of underlying TCP connections.
